### ISSUE TITLE
"Failed to make ast node `Name`"

### ISSUE THREAD

**Issue by SludgePhD** [rust-lang/rust-analyzer#18724]
**rust-analyzer version**: 0.3.2220-standalone

**rustc version**: current stable, but it seems like any should work

**editor or extension**: VSCode

**relevant settings**: diagnostics are enabled (by default)

**code snippet to reproduce**: 
Not sure what triggers this, sorry!

```
Document pull failed for text document file:///home/sludge/code/Artifice/crates/sludgedsp/src/backend/jack.rs
  Message: request handler panicked: Failed to make ast node `syntax::ast::generated::nodes::Name` from text mod self;

[...]

Failed to make ast node `syntax::ast::generated::nodes::Name` from text mod self;
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: syntax::ast::make::name
   3: ide_diagnostics::handlers::unresolved_field::unresolved_field
   4: ide_diagnostics::semantic_diagnostics
   5: ide_diagnostics::full_diagnostics
   6: ra_salsa::Cancelled::catch
   7: rust_analyzer::handlers::request::handle_code_action
   8: core::ops::function::FnOnce::call_once{{vtable.shim}}
```

In the same file that triggers the error, there is also a possibly related false positive syntax error that sticks around even after saving: `Syntax Error: expected SEMICOLON`. The semicolon in question most definitely exists, and the program builds and runs without issues (other than bugs I introduced myself).

**Comment by Veykril** [comment]
This is fixed on master, so next monday
